src/main/java/org/whispersystems/textsecuregcm/auth/AuthorizationTokenGenerator.java

```diff
@@ -58,6 +62,20 @@ public boolean isValid(String token, String number, long currentTimeMillis) {
     return isValidSignature(parts[0] + ":" + parts[1], parts[2]);
   }
 
+  private String getUserId(String number) {
+    if (userIdKey.isPresent()) {
+      try {
+        Mac mac = Mac.getInstance("HmacSHA256");
```

Comment:
getInstance() is kind of expensive, I'd reuse the same Mac instance inline

Reply:
I'm not sure I understand this comment. Can you explain in more detail ?

Reply:
I talked to Jeff and his interpretation was the same as my own.
I have a plan of attack and will make changes to reflect this feedback.

Reply:
done in 53d191103ccbe5ccd7699d44d3e9cd5469bc3b5a